executable-spec/src/Cardano/Ledger/Spec/STS/Update/Implementation.hs

```diff
@@ -23,11 +28,13 @@ import           Ledger.Core (Slot (Slot))
 import qualified Cardano.Ledger.Spec.STS.Update.Data as Data
 
 
-data IMPLEMENTATION
+data IMPLEMENTATION hashAlgo
 
-data Env =
+data Env hashAlgo =
   Env { currentSlot :: !Slot
         -- ^ The current slot in the blockchain system
+      , apprvsips :: !(Set (Data.SIPHash hashAlgo))
```

Comment:
I'm not sure I understand this field. My guess is that it can't record a result, but record the approved SIP's. Is that correct?

Reply:
And why did you added this field, to prepare `Env` for implementing the `IMPLEMENTATION` phase? Or is there any other fundamental reason why we need this?

Reply:
You are right. The field comment was wrong that is why you were confused. This set included all approved SIPs. In the implementation phase, only approved SIPs are allowed, so it is a must-have candidate for its environment. Yes it is just a small preparation step.

Reply:
Done OK